api/src/auth/auth.ts

```diff
@@ -1,14 +1,21 @@
 import type { SchemaOverview } from '@directus/types';
 import type { Knex } from 'knex';
+import { UsersService } from '../services/users.js';
 import type { AuthDriverOptions, User } from '../types/index.js';
+import { getSchema } from '../utils/get-schema.js';
 
 export abstract class AuthDriver {
 	knex: Knex;
-	schema: SchemaOverview;
 
 	constructor(options: AuthDriverOptions, _config: Record<string, any>) {
 		this.knex = options.knex;
-		this.schema = options.schema;
+	}
+
+	protected async getUsersService(schema?: SchemaOverview): Promise<UsersService> {
+		return new UsersService({
+			knex: this.knex,
+			schema: schema ?? (await getSchema()),
+		});
```

Comment:
I dont think we need this function to be `async` all usages (Except 1) get the schema passed in. This way we do not have to `await` this function everywhere for a fallback that isnt used 😄 
```suggestion
	protected getUsersService(schema: SchemaOverview): UsersService {
		return new UsersService({
			knex: this.knex,
			schema,
		});
```

Reply:
Nice suggestion :)